### PR TITLE
London 9 - Oleh Pysmenko - JS2 - Week 4

### PR DESCRIPTION
**Pull request by tiritaki**
# Your Details

- Your Name: Oleh Pysmenko
- Your City: London
- Your Slack Name: tiritaki

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,4 +1,33 @@
-function setAlarm() {}
+function setAlarm() {
+  let inputValue = document.getElementById('alarmSet')
+  let remainingTime = document.getElementById('timeRemaining')
+
+  let minutes = ''
+  let seconds = ''
+
+ let interval = setInterval(()=>{
+  inputValue.value--
```
Comment (nuclear-pasta) [inline 1-mandatory/1-alarmclock/alarmclock.js]: I think the inputValue should not change in the UI. Only the remaining time should.

In order to do this you could store the inputValue.value outside the interval as in:


```


let value = inputValue.value--
 let interval = setInterval(()=>{
  value--
...
}

```
and change all the references to inputValue to the new variable

File: 1-mandatory/3-slideshow/slideshow.js [hunk 2]
```diff
@@ -1 +1,22 @@
...
+
+function next () {
+    if (i >= images.length-1) i = -1
+    i++;
+    return setImg()
+}
+
+function setImg (){
+    return sliderImage.setAttribute('src', "assets/" + images[i])
+}
```
Comment (nuclear-pasta) [inline 1-mandatory/3-slideshow/slideshow.js:22]: Nice! If you want give a shot for the auto forward and back, you could use the same logic as the first exercise with a setInterval

File: 1-mandatory/1-alarmclock/alarmclock.js (hunk 1 above)
Comment (tiritaki) [reply 1-mandatory/1-alarmclock/alarmclock.js]: thank you for this advise, working as it should

File: 1-mandatory/3-slideshow/slideshow.js (hunk 2 above)
Comment (tiritaki) [reply 1-mandatory/3-slideshow/slideshow.js:22]: thank you for approach how to solve auto logic for slider. Hope I will have some time to sit on it.

**Pull request closed without merging**